component/client/ClientCore.jsx

```diff
@@ -0,0 +1,53 @@
+import React, {Component} from "react";
+import {Grid, Row} from "react-bootstrap";
+import ClientNavbar from "./ClientNavbar.jsx";
+
+class ClientCore extends Component {
+  constructor(props) {
+    super(props);
+
+    // setup components map.
+    // TODO: add lifecycle to go properly set state based on persistant use.
+    // TODO: create chat and setting components.
+    this.components = {
+      "chat": <h1> chat </h1>,
+      "setting": <h1> setting </h1>
+    };
+
+    // initialize the activeKey.
+    this.state = {
+      activeKey: "chat"
+    };
+
+    // bind helper functions.
+    this.handleSelection = this.handleSelection.bind(this);
```

Comment:
It is not defined in the global scope. It is defined in class scope and is standard in React to do so in the constructor if you have non-lifecycle methods.